From 2f9fc6fcd22f67a9b839e54d21008ba1623e662d Mon Sep 17 00:00:00 2001
From: Nigel Stewart <nigels@nigels.com>
Date: Sun, 29 Mar 2026 17:51:22 +1000
Subject: [PATCH 1/4] UPSTREAM: fix: GL_ARB_sparse_buffer support for
 GL_EXT_direct_state_access and GL_ARB_direct_state_access

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 auto/bin/filter_gl_ext.sh | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/auto/bin/filter_gl_ext.sh b/auto/bin/filter_gl_ext.sh
index 87bb2c8..0fa52ce 100755
--- a/auto/bin/filter_gl_ext.sh
+++ b/auto/bin/filter_gl_ext.sh
@@ -390,6 +390,18 @@ EOT
     void glNamedBufferStorageEXT (GLuint buffer, GLsizeiptr size, const void *data, GLbitfield flags)
 EOT
 
+# add missing functions to GL_EXT_direct_state_access (GL_ARB_sparse_buffer related)
+# https://github.com/nigels-com/glew/issues/470
+    cat >> $1/GL_EXT_direct_state_access <<EOT
+    void glNamedBufferPageCommitmentEXT (GLuint buffer, GLintptr offset, GLsizeiptr size, GLboolean commit)
+EOT
+
+# add missing functions to GL_ARB_direct_state_access (GL_ARB_sparse_buffer related)
+# https://github.com/nigels-com/glew/issues/470
+    cat >> $1/GL_ARB_direct_state_access <<EOT
+    void glNamedBufferPageCommitmentARB (GLuint buffer, GLintptr offset, GLsizeiptr size, GLboolean commit)
+EOT
+
 # Filter out GL_UNSIGNED_INT and GL_FLOAT from GL_AMD_performance_monitor
     grep -v 'GL_UNSIGNED_INT ' $1/GL_AMD_performance_monitor > tmp
     mv tmp $1/GL_AMD_performance_monitor
-- 
2.52.0

